src/main/java/seedu/taassist/model/TaAssist.java

```diff
@@ -162,20 +163,36 @@ public void removeModuleClass(ModuleClass moduleClass) {
 
 
     /**
-     * Removes the {@code session} from the {@code moduleClass}
-     * as well as all students in the {@code moduleClass}.
+     * Removes the {@code sessions} from the {@code moduleClass} as well as all students in the {@code moduleClass} and
+     * returns the new module class with the sessions removed.
      */
-    public void removeSession(ModuleClass moduleClass, Session session) {
+    public ModuleClass removeSessions(ModuleClass moduleClass, Set<Session> sessions) {
+        requireAllNonNull(moduleClass, sessions);
+        for (Session session: sessions) {
+            // Update student data
+            removeSessionFromStudents(moduleClass, session);
+            // Update module class data
+            moduleClass = removeSessionFromModuleClass(moduleClass, session);
+        }
+        return moduleClass;
+    }
+
+    private void removeSessionFromStudents(ModuleClass moduleClass, Session session) {
         requireAllNonNull(moduleClass, session);
         List<Student> updatedStudents = students.asUnmodifiableObservableList().stream()
                 .map(student -> student.removeSession(moduleClass, session))
                 .collect(Collectors.toList());
         setStudents(updatedStudents);
-        setModuleClass(moduleClass, moduleClass.removeSession(session));
     }
 
-    //// util methods
+    private ModuleClass removeSessionFromModuleClass(ModuleClass moduleClass, Session session) {
+        requireAllNonNull(moduleClass, session);
+        ModuleClass newModuleClass = moduleClass.removeSession(session);
+        setModuleClass(moduleClass, newModuleClass);
```

Comment:
Would it work if ` setModuleClass(moduleClass, newModuleClass);` is done once in the end? Then this function won't be needed in `removeSessions`.

Reply:
yea it works!